USFMToolsSharp.Renderers.Docx/DocxRenderer.cs

```diff
@@ -0,0 +1,288 @@
+                            footnoteId = $"{FootnoteTextTags.Count + 1}";
+                            break;
+                        default:
+                            footnoteId = fMarker.FootNoteCaller;
+                            break;
+                    }
+                    XWPFRun footnoteMarker = parentParagraph.CreateRun();
+
+                    footnoteMarker.SetText(footnoteId);
+                    footnoteMarker.Subscript = VerticalAlign.SUPERSCRIPT;
+
+                    FootnoteTextTags[footnoteId] = fMarker;
+
+                    break;
+                case FTMarker fTMarker:
+
+                    foreach (Marker marker in input.Contents)
+                    {
+                        RenderMarker(marker,parentParagraph);
+                    }
+
+                    break;
+                case FQAMarker fQAMarker:
+                    foreach (Marker marker in input.Contents)
+                    {
+                        RenderMarker(marker, parentParagraph,isItalics:true);
+                    }
+                    break;
+                case FQAEndMarker fQAEndMarker:
+                case FEndMarker _:
+                case IDEMarker _:
+                case IDMarker _:
+                case VPMarker _:
+                case VPEndMarker _:
+                    break;
+                default:
+                    UnrenderableTags.Add(input.Identifier);
+                    break;
+            }
+        }
+        private void RenderFootnotes()
+        {
+
+            if (FootnoteTextTags.Count > 0)
+            {
+                XWPFParagraph renderFootnoteHeader = newDoc.CreateParagraph();
+                XWPFRun FootnoteHeader = renderFootnoteHeader.CreateRun();
+                FootnoteHeader.SetText("Footnotes");
+                FootnoteHeader.FontSize = 24;
+
+                foreach(KeyValuePair<string,Marker> footnoteKVP in FootnoteTextTags)
+                {
+                    XWPFParagraph renderFootnote = newDoc.CreateParagraph();
+                    XWPFRun footnoteMarker = renderFootnote.CreateRun();
+                    footnoteMarker.SetText(footnoteKVP.Key);
+                    footnoteMarker.Subscript = VerticalAlign.SUPERSCRIPT;
+
+                    foreach(Marker marker in footnoteKVP.Value.Contents)
+                    {
+                        RenderMarker(marker, renderFootnote);
+                    }
+  
+                }
+                FootnoteTextTags.Clear();
+            }
+        }
+        public void setDocLineSpacing(int num)
+        {
+            foreach (XWPFParagraph par in newDoc.Paragraphs)
+            {
+                par.SpacingBetween = num * 240;
+            }
+        }
+        public void setDocTextAlignment(ParagraphAlignment align)
+        {
+            foreach (XWPFParagraph par in newDoc.Paragraphs)
+            {
+                par.Alignment = align;
+            }
+        }
+        public void setDocColumnCount(int num)
```

Comment:
This also should be config and not a function because it also won't change after render